src/schemas/utils.ts

```diff
@@ -77,6 +78,36 @@ const convertClinicalFileErrorToGql = (fileError: {
   };
 };
 
+const convertRegistrationDataToGql = (
+  programShortName: string,
+  data: {
+    registration: DeepReadonly<ActiveRegistration> | undefined;
+    errors?: RegistrationErrorData[];
+    batchErrors?: { message: string; batchNames: string[]; code: string }[];
+  },
+) => {
+  const registration = get(data, 'registration', {} as Partial<typeof data.registration>);
+  const schemaAndValidationErrors = get(data, 'errors', [] as typeof data.errors);
+  const fileErrors = get(data, 'batchErrors', [] as typeof data.batchErrors);
```

Comment:
using annotations is better than assertions
assertions is basically escape hatching the TS check by telling TS you know the type better than it knows it

lots of our legacy code uses assertions as the way we used a type system
in general this is not correct usage

Reply:
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-annotations-on-variables
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions

Reply:
Updated! Generally left the legacy code as is, good opportunity to improve